src/uploadSarif.ts

```diff
@@ -2,35 +2,41 @@ import { Octokit } from "@octokit/core";
 import * as github from "@actions/github";
 import * as zlib from "zlib";
 import * as ghCore from "@actions/core";
-import * as fs from "fs";
+import { URLSearchParams } from "url";
+import { promises as fs } from "fs";
+
+import { promisify } from "util";
 import * as utils from "./util/utils";
-import Crda from "./crda";
 
 export async function uploadSarifFile(
-    ghToken: string, sarifToUpload: string, checkoutPath: string,
-    analysisStartTime: string, ref?: string, sha?: string
+    ghToken: string, sarifToUpload: string, /* resolvedManifestPath: string, */
+    analysisStartTime: string, sha: string, ref: string,
 ): Promise<void> {
-    const sarifContents = fs.readFileSync(sarifToUpload, "utf-8");
-    ghCore.debug(`Raw upload size: ${sarifContents.length} bytes`);
-    const zippedSarif = zlib.gzipSync(sarifContents).toString("base64");
-    ghCore.info(`Zipped upload size: ${zippedSarif.length} bytes`);
+    const { owner, repo } = github.context.repo;
+    ghCore.info(`⬆️ Uploading SARIF file to ${owner}/${repo}...`);
+
+    const sarifContents = await fs.readFile(sarifToUpload, "utf-8");
+    ghCore.debug(`Raw upload size: ${utils.convertToHumanFileSize(sarifContents.length)}`);
+    const zippedSarif = (await promisify(zlib.gzip)(sarifContents)).toString("base64");
+    ghCore.info(`Zipped upload size: ${utils.convertToHumanFileSize(zippedSarif.length)}`);
```

Comment:
```suggestion
    ghCore.debug(`Zipped upload size: ${utils.convertToHumanFileSize(zippedSarif.length)}`);
```

Reply:
I like this as info level. In case it's large the upload would be slow.

Reply:
Okay :+1: